ledger-core/v2/virtual/execute/execute_test.go

```diff
@@ -156,14 +156,104 @@ func TestSMExecute_StartRequestProcessing(t *testing.T) {
 	assert.Equal(t, smObject.KnownRequests.Len(), 1)
 	assert.True(t, smObject.KnownRequests.GetList(contract.CallTolerable).Exist(smExecute.execution.Outgoing))
 
-	{ // update known requests panics
+	{
+		// call same request
+		// expect duplicate detection and SM stop
+		wasStoped := false
+
 		execCtx := smachine.NewExecutionContextMock(mc).
-			UseSharedMock.Set(shareddata.CallSharedDataAccessor)
+			UseSharedMock.Set(shareddata.CallSharedDataAccessor).
+			LogMock.Return(smachine.Logger{}).
+			StopMock.Set(
+			func() (s1 smachine.StateUpdate) {
+				wasStoped = true
+				return smachine.StateUpdate{}
+			})
```

Comment:
No need to move that check `assert.Equal(t, true, wasStoped)` from here, the presence of that mock function + `mc.Finish()` - guarantee that it was stopped.
```
func EmptyStop() func() smachine.StateUpdate {
    return func() smachine.StateUpdate {
		return smachine.StateUpdate{}
	}	
}

....

smachine.NewExecutionContextMock(mc).
    StopMock.Set(EmptyStop())
```

Reply:
or simple `StopMock.Return(smachine.StateUpdate{})`